stdlib/public/core/Integers.swift.gyb

```diff
@@ -727,3 +776,440 @@ extension SignedInteger_ where Self : FixedWidthInteger {
+}
+
+
+// Create an ambiguity when indexing or slicing
+// Range[OfStrideable]<${Self}> outside a generic context.  See
+// Range.swift for details.
+extension ${Self} {
+  public typealias _DisabledRangeIndex = ${Self}
+}
+
+
+% for src_type in all_integer_types(word_bits):
+%    srcBits = src_type.bits
+%    srcSigned = src_type.is_signed
+%    Src = src_type.stdlib_name
+// FIXME(integers): consider merging with
+//                  FixedWidthInteger.init(_truncatingBits:)
+%   if should_define_truncating_bit_pattern_init(src_ty=src_type, dst_ty=self_type):
+extension ${Self} {
+  /// Construct a `${Self}` having the same bitwise representation as
+  /// the least significant bits of the provided bit pattern.
+  ///
+  /// No range or overflow checking occurs.
+  @_transparent
+  public init(truncatingBitPattern: ${Src}) {
+    let src = truncatingBitPattern._value
+%     if self_type.bits == src_type.bits:
+    let dstNotWord = src
+%     else:
+    let dstNotWord = Builtin.trunc_Int${srcBits}_Int${bits}(src)
+%     end
+    self._value = dstNotWord
+  }
+}
+%   end
+% end
+
+% if signed:
+// TODO: Consider removing the underscore.
+/// Returns the argument and specifies that the value is not negative.
+/// It has only an effect if the argument is a load or call.
+@_transparent
+public func _assumeNonNegative(_ x: ${Self}) -> ${Self} {
+  _sanityCheck(x >= 0)
+  return ${Self}(Builtin.assumeNonNegative_${BuiltinName}(x._value))
+}
+% end
+
+
+// FIXME(integers): this conformance should be removed along with the
+// BitwiseOperations protocol
+@_transparent
+extension ${Self} : BitwiseOperations {
+  /// The empty bitset of type `${Self}`.
+  public static var allZeros: ${Self} { return 0 }
+}
+//===--- end of FIXME(integers) -------------------------------------------===//
+
+//===----------------------------------------------------------------------===//
+// FIXME(integers): Remove these operators in favour of the generic ones on
+//                  Arithmetic
+//===----------------------------------------------------------------------===//
+// Operations with potentially-static overflow checking
+//
+// FIXME: must use condfail in these operators, rather than
+// overflowChecked, pending <rdar://problem/16271923> so that we don't
+// foil static checking for numeric overflows.
+% for op, method, label in [
+%    ('+', 'adding', ''),
+%    ('*', 'multiplied', 'by: '),
+%    ('-','subtracting', ''),
+%    ('/', 'divided', 'by: '),
+%    ('%', 'remainder', 'dividingBy: ')
+%  ]:
+@_transparent
+public func ${op} (lhs: ${Self}, rhs: ${Self}) -> ${Self} {
+  return lhs.${method}(${label}rhs)
+}
+% end
+//===--- end of FIXME(integers) -------------------------------------------===//
+
+% end # end of concrete FixedWidthInteger section
+
+
+// FIXME(integers): inline manually everywhere
+public func numericCast<T : BinaryInteger, U : BinaryInteger>(_ x: T) -> U {
+  return U(x)
+}
+
+internal func _unsafePlus(_ lhs: Int, _ rhs: Int) -> Int {
```

Comment:
They are used elsewhere, and I just copied them from the old implementation to make the whole thing compile. The plan is to replace them with the method call, since we have the unsafeOP methods on `FixedWidthInteger`.